Scripts/UI/GradientButton.cs

```diff
@@ -321,19 +385,16 @@ IEnumerator BeginHighlight()
 			this.StopCoroutine(ref m_IconHighlightCoroutine);
 			m_IconHighlightCoroutine = StartCoroutine(IconContainerContentsBeginHighlight());
 
-			const float kTargetTransitionAmount = 1f;
 			var transitionAmount = Time.deltaTime;
-			var shapedTransitionAmount = 0f;
 			var currentGradientPair = GetMaterialColors();
 			var targetGradientPair = highlightGradientPair;
 			var currentLocalScale = m_ContentContainer.localScale;
-			var highlightedLocalScale = new Vector3(m_OriginalContentContainerLocalScale.x, m_OriginalContentContainerLocalScale.y, m_OriginalContentContainerLocalScale.z * m_highlightZScaleMultiplier);
-			while (transitionAmount < kTargetTransitionAmount)
+			var highlightedLocalScale = highlightScale;
+			var highlightDuration = m_BeginHighlightDuration > 0f ? m_BeginHighlightDuration : 0.01f;  // Add sane default if highlight duration is zero
```

Comment:
I'd go with a warning in Awake or something. What happens if the duration is 0? Wouldn't it just skip the while loop and immediately set the target color?

Reply:
Addressed, the look is just skipped now with a zero (or less) duration